docs/Features.md

```diff
@@ -1,49 +1,122 @@
+
+* surface reference API (DEPRECATED in CUDA)
+
+* surface object API (hipCreateSurfaceObject, hipDestroySurfaceObject)
+
+* peer access (hipMemcpyPeer, hipMemcpyPeerAsync,
   hipDeviceCanAccessPeer, hipDeviceEnablePeerAccess,
-  hipDeviceDisablePeerAccess, hipDeviceGetStreamPriorityRange,
-  hipDevicePrimaryCtxRelease, hipDevicePrimaryCtxRetain,
-  hipDevicePrimaryCtxSetFlags, hipMemPrefetchAsync, hipMemAdvise,
-  hipModuleLoadData, hipModuleUnload, hipModuleLaunchKernel
+  hipDeviceDisablePeerAccess, etc)
+
+* hipMemRangeGetAttribute
+
+* some config APIs (hipDeviceSetCacheConfig, hipDeviceGetCacheConfig
+  hipDeviceSetSharedMemConfig, hipDeviceGetSharedMemConfig,
+  hipSetDeviceFlags, hipGetDeviceFlags, hipFuncSetCacheConfig)
+
+* primary context API (hipDevicePrimaryCtxRelease,
+  hipDevicePrimaryCtxRetain,  hipDevicePrimaryCtxSetFlags)
+
+* few memory APIs (hipMemPrefetchAsync, hipMemAdvise)
+
+* few module APIs (hipModuleLoadData, hipModuleUnload, hipModuleLaunchKernel)
 
 #### partially supported
+
   * Texture Objects of 1D/2D type are supported; 3D, LOD, Grad,
     Cubemap, Gather and Mipmapped textures are not supported
 
+  * hiprtc: Referring global device variables, constants and texture
+    references in the name expressions are not supported.
+
+  * hipFuncGetAttributes - not all attributes are supported, depends on backend
+
+  * hipDeviceGetLimit - only some limits are supported
+
+-------------------------------------------------------------------
+
+
 ### Device side
 
 #### Unsupported / unavailable
-* __syncwarp(), __activemask()
-* cooperative_groups.h header
-* Address Space Predicate Functions, Address Space Conversion Functions
-* alloca(), malloc(), free()
-* Warp Reduce Functions, Warp Matrix Functions
+
+* _syncwarp(), __activemask(), threadfence_system(), warpSize variable
+
+* clock(), clock64()
+
+* Cooperative Groups Functions
+
+* Warp Matrix Functions
+
+* Independent Thread Scheduling
+
+* Profiler Counter Function
+
+* Address Space Predicate Functions
+
+* Address Space Conversion Functions
+
+* Device-Side Dynamic Global Memory Allocation
+
+* In-Line Assembly
+
+* __trap(), __brkpt(), assert()
+
+* surface functions
 
 #### Partially supported
 
+* math library: almost all single/double functions are available,
+  half/half2 functions are available but untested
+
+* functions that specify rounding (e.g. __fdiv_rd), except conversion functions (as rounding cannot be selected in OpenCL)
+
+* device intrinsics are mapped to OpenCL's native_* functions where possible, otherwise are unsupported / emulated
+
 * Warp functions (__all, __any, __ballot): only the non-sync versions are supported
+
 * Shuffle functions (__shfl_{up,down,xor}): only the non-sync versions are supported
-* assert(), __trap, __brkpt are not available but abort() is
-* mathematical library: almost all single/double functions are available,
-  but half-precision variants are not available
+
+* abort()
```

Comment:
Isn't this __trap()?

Reply:
it might be called `__trap` in CUDA, but HIP docs talks about `abort` and CHIP-SPV code also contains `abort`:

```
static inline __device__ void abort() {
  __chipspv_abort(&__chipspv_abort_called);
}
```

do we want to rename it ?

Reply:
OK, I've missed that. No, we should support HIP, but hopefully we also have a __trap() that calls abort() too :)